src/plugins/intel_gpu/include/intel_gpu/graph/program.hpp

```diff
@@ -249,8 +250,14 @@ struct program {
     std::pair<int64_t/*const alloc*/, int64_t/*general alloc*/> get_estimated_device_mem_usage();
 
     void remove_kernel(kernel_id id);
-    void calc_nodes_hash();
 
+    struct ImplHasher {
```

Comment:
Does it not make sense to make this structure a part of kernel_impl_params.hpp?

Reply:
Makes sense. Will update it in a next PR